fw/sock_clnt.c

```diff
@@ -342,12 +346,42 @@ tfw_h2_sk_prepare_xmit(struct sock *sk, struct sk_buff *skb,
 		 * We clear this flag to prevent it's copying
 		 * during skb splitting.
 		 */
-		if (!stream->xmit.b_len) {
+		if (!stream->xmit.t_len && !stream->xmit.b_len) {
```

Comment:
Why? If body is finished we should process stream and clear SS_F_HTTT2_FRAME_DATA. It is not depend on trailers

Reply:
No, here it's used to avoid stream state machine processing, which assumes DATA is the last step, but if there is trailers, the last step is trailers, so it should be postponed.

Reply:
Stream state machine should assumes that DATA s not a last step. We should rework it to  make it possible to handle trailers as I think (when we receive trailers it is already done). @krizhanovsky what do you think?